src/manifest_store.rs

```diff
@@ -91,6 +91,16 @@ impl FenceableManifest {
         self.stored_manifest.new_checkpoint(checkpoint_id, options)
     }
 
+    pub(crate) async fn write_checkpoint(
+        &mut self,
+        checkpoint_id: Option<Uuid>,
+        options: &CheckpointOptions,
+    ) -> Result<Checkpoint, SlateDBError> {
+        self.stored_manifest
```

Comment:
Do we need to `check_epoch()?` here?

Reply:
ah yeah we do. I forgot that `write_checkpoint` writes the manifest in a refresh/apply loop.

Reply:
fixed, and added a test